mlir/include/mlir/Dialect/Polynomial/IR/Polynomial.td

```diff
@@ -123,19 +145,20 @@ def Polynomial_RingAttr : Polynomial_Attr<"Ring", "ring"> {
   let parameters = (ins
     "Type": $coefficientType,
     OptionalParameter<"::mlir::IntegerAttr">: $coefficientModulus,
-    OptionalParameter<"::mlir::polynomial::PolynomialAttr">: $polynomialModulus,
+    OptionalParameter<"::mlir::polynomial::IntPolynomialAttr">: $polynomialModulus,
     OptionalParameter<"::mlir::IntegerAttr">: $primitiveRoot
   );
-
+  let assemblyFormat = "`<` struct(params) `>`";
   let builders = [
     AttrBuilder<
```

Comment:
Seems like this could be a `AttrBuilderWithInferredContext` instead since it takes Type/Attr operands already.

Reply:
Fixed in 1230b731e68c